test/e2e/user_workload_monitoring_test.go

```diff
@@ -965,227 +561,422 @@ func assertTenancyForRules(t *testing.T) {
+					Name: "example",
+					Rules: []monitoringv1.Rule{
+						{
+							Record: "version:blah:count",
+							Expr:   intstr.FromString(`count(version)`),
+						},
+						{
+							Alert: "VersionAlert",
+							Expr:  intstr.FromString(fmt.Sprintf(`version{namespace="%s",job="prometheus-example-app"} == 1`, userWorkloadTestNs)),
+							For:   "1s",
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
 
-			if resp.StatusCode/100 == 2 {
-				return fmt.Errorf("unexpected status code response, got %d (%s)", resp.StatusCode, framework.ClampMax(b))
-			}
+	_, err = f.MonitoringClient.PrometheusRules(userWorkloadTestNs).Create(ctx, &monitoringv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "prometheus-example-rule-leaf",
+			Labels: map[string]string{
+				"k8s-app": "prometheus-example-rule-leaf",
+				"openshift.io/prometheus-rule-evaluation-scope": "leaf-prometheus",
+			},
+		},
+		Spec: monitoringv1.PrometheusRuleSpec{
+			Groups: []monitoringv1.RuleGroup{
+				{
+					Name: "example",
+					Rules: []monitoringv1.Rule{
+						{
+							Record: "version:blah:leaf:count",
+							Expr:   intstr.FromString(`count(version)`),
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
 
-			return nil
-		})
-		if err != nil {
-			t.Fatalf("the account has access to the %q endpoint of Thanos querier: %v", path, err)
-		}
+	err = f.OperatorClient.WaitForDeploymentRollout(ctx, app)
+	if err != nil {
+		return err
 	}
+	return nil
 }
 
-func assertPrometheusAlertmanagerInUserNamespace(t *testing.T) {
+func createPrometheusAlertmanagerInUserNamespace(t *testing.T) error {
+	t.Helper()
 	ctx := context.Background()
-	_, err := f.KubeClient.AppsV1().StatefulSets(userWorkloadTestNs).Get(ctx, "prometheus-not-to-be-reconciled", metav1.GetOptions{})
-	if err == nil {
-		t.Fatal("expected no Prometheus statefulset to be deployed, but found one")
+	_, err := f.MonitoringClient.Alertmanagers(userWorkloadTestNs).Create(ctx, &monitoringv1.Alertmanager{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "not-to-be-reconciled",
+		},
+		Spec: monitoringv1.AlertmanagerSpec{
+			Replicas: proto.Int32(1),
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
 	}
 
-	_, err = f.KubeClient.AppsV1().StatefulSets(userWorkloadTestNs).Get(ctx, "alertmanager-not-to-be-reconciled", metav1.GetOptions{})
-	if err == nil {
-		t.Fatal("expected no Alertmanager statefulset to be deployed, but found one")
+	_, err = f.MonitoringClient.Prometheuses(userWorkloadTestNs).Create(ctx, &monitoringv1.Prometheus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "not-to-be-reconciled",
+		},
+		Spec: monitoringv1.PrometheusSpec{
+			Replicas: proto.Int32(1),
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
 	}
+	return nil
 }
 
-func assertGRPCTLSRotation(t *testing.T) {
+// cleanupUWM tears down or fails the test that calls it
+func cleanupUWM(t *testing.T) {
```

Comment:
what do you think about move `setupUWM` and `cleanupUWM` to the framework directory?

Reply:
done